url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/util/SemanticSkeletonGenerator.java

```diff
@@ -0,0 +1,226 @@
+package fri.patterns.interpreter.parsergenerator.util;
+
+import java.util.*;
+import java.io.*;
+import fri.patterns.interpreter.parsergenerator.syntax.*;
+
+/**
+	This class can be called using SourceGenerator by commandline.
+	<p>
+	Generates Java code that contains empty method bodies for the
+	implementation of a given Syntax. For every rule there will
+	be a method with the name of the nonterminal on the left side
+	and one argument for each symbol on the right side. All arguments
+	are of type Object.
+	The created implementation derives <i>ReflectSemantic</i>.
+	<pre>
+		File in = new File("MySyntax.grammar);
+		String basename = "MySemantic";
+		Writer out = new FileWriter(basename+".java");
+		new SemanticSkeletonGenerator(in, basename, "my.pkg.name", out);
+		// the Java source "my/pkg/name/MySemantic.java" will be generated
+	</pre>
+	
+	@see fri.patterns.interpreter.parsergenerator.semantics.ReflectSemantic
+	@see fri.patterns.interpreter.parsergenerator.util.SourceGenerator
+	@author (c) 2002, Fritz Ritzberger
+*/
+
+public class SemanticSkeletonGenerator
+{
+	private static Hashtable keyWords;
+	private BufferedWriter bw;
+	
+	/**
+		Reads the passed Syntax and turns it into a semantic skeleton implementation.
+		The passed Write will be closed.
+		@param syntax Syntax the semantc is meant for
+		@param className basname of the class to generate (without package)
+		@param pkgName name of package, can be null
+		@param skeletonOutput Writer where Java source should be written to
+	*/
+	public SemanticSkeletonGenerator(Syntax syntax, String className, String pkgName, Writer skeletonOutput)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.